### PR TITLE
Added MathUtils::poly

### LINKED ISSUE #13184: Polynomial MathUtils
## Reason
Make polynomial method in MathUtils for quick and easy polynomial evaluations.

## Design
Use Horner's method to calculate polynomial. Provide easy derivative calculation as well.

## Impact
Simpler code.

### PR DESCRIPTION
**Pull request by tophmatthews**
Closes #13184

### CODE REVIEW

File: framework/src/utils/MathUtils.C [hunk 1]
```diff
@@ -235,4 +235,28 @@ pow(Real x, int e)
   return neg ? 1.0 / result : result;
 }
 
+// Evaluate value of polynomial using Horner's method
+Real
+poly(std::vector<Real> c, const Real x, const bool derivative)
+{
+  const Real size = c.size();
```
Comment (dschwen) [inline framework/src/utils/MathUtils.C]: I'd make `size` and integer (after all below you are doing integer math with it). I would also suggest to template this  on the number type to make it usable with AD in teh future.

Comment (tophmatthews) [reply framework/src/utils/MathUtils.C]: Whoa...you read my mind about templating without me having to ping you...

Comment (tophmatthews) [reply framework/src/utils/MathUtils.C]: Probably should do it to pow too.

**Comment by YaqiWang** [comment]
How about create a class `Polynomial`, which is constructed from a vector of Real and can have member functions like `(x)`, `derivative(x)`, `operator*(x)`, `operator*(Polynomial)`, etc.?

**Comment by tophmatthews** [comment]
Go for it! :)

**Review by dschwen — APPROVED** [review]
looks good to me

**Pull request merged**